### PR TITLE
Add joystick to servo generic component

### PR DESCRIPTION
**Pull request by destogl**
### Description

The current demonstration component `JoyToServo` is tailor-made for the Panda manipulator, going so far that the joint names are hard-coded. It also does not provide control of all joints and all cartesian directions.

So in this PR, I propose a solution using a coupling with [ros2/teleop_twist_joy](https://github.com/ros2/teleop_twist_joy) package for the cartesian control and implementing only joint-based control using a joystick. The implementation is currently tested with a non-public manipulator using FakeSystem (#361) implementation in ros2_control. I added only a configuration file for the [Xbox 360 Wired](https://wiki.ros.org/joy?distro=noetic#Microsoft_Xbox_360_Wired_Controller_for_Linux) joystick since this is the one currently available to me.

For now, this PR draft to share my work, which could be useful for someone. Before making it a proper PR, the following need to be done:

- [ ] Add a section about this node in the documentation
- [ ] Add support for FakeSystem in Panda URDF - partially from (#361)
- [ ] Update example launch file
- [ ] Test example
- [ ] Update documentation

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/moveit_servo/config/joystick_to_servo.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  - joint1
+  - joint2
+  - joint3
+  - joint4
+  - joint5
+  - joint6
+  - joint7
+
+# Do not change this names. The joints are order as in the list above
+num_joint_axis_control_modes: 2
```
Comment (AndyZe) [inline moveit_ros/moveit_servo/config/joystick_to_servo.yaml:26]: I think you could use a comment here. This confused me initially:

`# Each joystick axis has up to 2 different behaviors, depending on the control mode`

File: moveit_ros/moveit_servo/src/teleop_demo/joystick_to_servo.cpp [hunk 2]
```diff
@@ -0,0 +1,198 @@
+/*********************************************************************
```
Comment (AndyZe) [inline moveit_ros/moveit_servo/src/teleop_demo/joystick_to_servo.cpp:1]: I'm glad you put this in its own `teleop_demo` folder

**Review by AndyZe — COMMENTED** [review]
At a high level, I like what this PR does. It goes above and beyond what you can do with teleop_tools. So I would support merging it when it works with the Panda.

File: moveit_ros/moveit_servo/config/joystick_to_servo.yaml (hunk 1 above)
Comment (destogl) [inline moveit_ros/moveit_servo/config/joystick_to_servo.yaml:26]:
```suggestion
# Each joystick axis has up to 2 different behaviors, depending on the control mode
num_joint_axis_control_modes: 2 
```

**Comment by AndyZe** [comment]
After some more thought, I'm not sure this really belongs in the Servo package. We already have the `teleop_tools` method here as well as the `joystick_to_servo_example.cpp` (a very basic demo.

What would you think about moving this to a different repo? I created [this one ](https://github.com/PickNikRobotics/xbox_translator) and was thinking about porting it there.

File: moveit_ros/moveit_servo/config/xbox360_wired.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+axis_linear:
```
Comment (AndyZe) [inline moveit_ros/moveit_servo/config/xbox360_wired.yaml:1]: One of the questions I have when looking at this is:  Why are there 2 config files? It seems like `joystick_to_servo.yaml` is for joint commands, `xbox360_wired.yaml` is for Cartesian commands?

**Pull request closed without merging**